modules/core/src/mean.dispatch.cpp

```diff
@@ -130,13 +130,47 @@ Scalar mean(InputArray _src, InputArray _mask)
     CV_Assert( mask.empty() || mask.type() == CV_8U );
 
     int k, cn = src.channels(), depth = src.depth();
-    Scalar s;
+    Scalar s = Scalar::all(0.0);
+
+    CV_Assert( cn <= 4 );
 
     CV_IPP_RUN(IPP_VERSION_X100 >= 700, ipp_mean(src, mask, s), s)
 
+    if (src.isContinuous() && mask.isContinuous())
+    {
+        int status = cv_hal_meanStdDev(src.data, 0, (int)src.total(), 1, src.type(),
```

Comment:
Is it possible to use CALL_HAL_RET macro here and below?

Reply:
The macro expects, that the return value is the last HAL function parameter. It's not applicable here.

Reply:
Perhaps we should add a new macro which doesn't pass `retval` to HAL function:
```.cpp
// name TBD
#define CALL_HAL_RET2(name, fun, retval, ...) \
    int res = __CV_EXPAND(fun(__VA_ARGS__)); \
    if (res == CV_HAL_ERROR_OK) \
        return retval; \
    else if (res != CV_HAL_ERROR_NOT_IMPLEMENTED) \
        CV_Error_(cv::Error::StsInternal, \
            ("HAL implementation " CVAUX_STR(name) " ==> " CVAUX_STR(fun) " returned %d (0x%08x)", res, res));
```
Then old macros could be redefined as follows:
```
#define CALL_HAL_RET(name, fun, retval, ...) \
    CALL_HAL_RET2(name, fun, retval, __VA_ARGS__, &retval)

#define CALL_HAL(name, fun, ...) \
    CALL_HAL_RET2(name, fun, ,__VA_ARGS__)

```

Reply:
Applied.